elections/uk/views/frontpage.py

```diff
@@ -79,7 +79,16 @@ def sopn_progress_by_election(self, election_slug=None, election_qs=None):
             return context
 
         context['posts_total'] = pe_qs.count()
-        context['posts_locked'] = pe_qs.filter(postextraelection__candidates_locked=True).count()
+        if election_qs:
+            pe_qs = pe_qs.filter(
+                postextraelection__candidates_locked=True,
+                postextraelection__election__in=election_qs
+            )
+        else:
+            pe_qs = pe_qs.filter(
+                postextraelection__candidates_locked=True,
```

Comment:
I think that for similar reasons this one will need an extra condition too: `postextraelection__election__slug__starswith=election_slug`. (I have no idea why that slug comparison is a `startswith` but it's best to be consistent anyway, I guess!)